### DISCUSSION TITLE
How to separate column definitions in FluentDataGrid

### DISCUSSION THREAD

**Discussion by kambet1** [microsoft/fluentui-blazor#1754] [Q&A] [answered]
Is there a way to separate column definition into another component?
The code below doesn't work.

```
<FluentDataGrid Items="@("ABC".AsQueryable())">
    <PropertyColumn Title="Column1" Property="row => row" />
    <PropertyColumn Title="Column2" Property="row => row" />
    <CommonColumns />
</FluentDataGrid>
```
```
@* CommonColumns.razor *@
<PropertyColumn Title="Column3" Property="(char row) => row" />
<TemplateColumn Title="Column4" TGridItem="char">@context</TemplateColumn>
```
thanks

**Comment by vnbaaij** [comment, marked as answer]
Hi. Sorry, but this is not possible with the Datagrid. It has to do with the way the grid uses ChildContent as a way of collecting the list of columns. They therefore need to be direct descendants.

**Reply by kambet1** [reply]
I understand.
Thank you for your quick reply.
I would be happy if I could do it somehow in the future.
